File: src/java/leetcode/editor/en/P1781_SumOfBeautyOfAllSubstrings.java
package java.leetcode.editor.en;
import java.util.*;
import java.util.function.*;
import java.leetcode.editor.en.definition.*;

public class P1781_SumOfBeautyOfAllSubstrings {
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int beautySum(String s) {
        int ans = 0;
        for (int i = 0; i < s.length(); i++) {
            int[] freq = new int[26];
            TreeMap<Integer, Integer> freqCount = new TreeMap<>();
            for (int j = i; j < s.length(); j++) {
                int f = freq[s.charAt(j) - 'a']++;
                freqCount.putIfAbsent(f+1, 0);
                freqCount.put(f+1, freqCount.get(f+1) + 1);
                if (f > 0) {
                    freqCount.put(f, freqCount.get(f) - 1);
                    if (freqCount.get(f) == 0) freqCount.remove(f);
                }
                ans += freqCount.lastKey() - freqCount.firstKey();
            }
        }
        return ans;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
}

File: src/kt/leetcode/editor/en/P1781_SumOfBeautyOfAllSubstrings.kt
package kt.leetcode.editor.en;
import java.util.*;
import java.util.function.*;
import kt.leetcode.editor.en.definition.*;
import kotlin.collections.HashMap

public class P1781_SumOfBeautyOfAllSubstrings {
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    fun beautySum(s: String): Int {
        var ans = 0
        for (i in 0 until s.length) {
            val count = IntArray(26)
            for (j in i until s.length) {
                count[s[j]-'a']++
                var max = Int.MIN_VALUE
                var min = Int.MAX_VALUE
                for (x in count) {
                    if (x == 0) continue
                    max = Math.max(max, x)
                    min = Math.min(min, x)
                }
                ans += max - min
            }
        }
        return ans
    }
}
//leetcode submit region end(Prohibit modification and deletion)
}
